contracts/protocol/ProtocolImplementation.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import "./ScoutTokenERC20.sol";
+import "./libs/MemoryUtils.sol";
+import "./libs/ProtocolAccessControl.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/StorageSlot.sol";
+
+contract ScoutProtocolImplementation is Context, ProtocolAccessControl {
+    using MemoryUtils for bytes32;
+
+
+    modifier onlyAdminOrClaimManager() {
+      require(_hasRole(MemoryUtils.CLAIM_MANAGER_SLOT) || _isAdmin() , "Proxy: caller is not the claim manager");
+        _;
+    }
+
+    // Allow the sender to claim their balance as ERC20 tokens
+    function claim(
+        string memory week,
```

Comment:
Keep the protocol simple by only giving it the notion of weeks

Seasons are managed by the caller